elitzur-scio/src/test/scala/com/spotify/elitzur/ValidatorDoFnTest.scala

```diff
@@ -80,4 +115,35 @@ class ValidatorDoFnTest extends PipelineSpec {
         .count
     }) shouldBe Seq(1)
   }
+
+  "Validator SCollection" should "validate collection of Seq,Vector,List or Array " +
+    "and set counters" in {
+    JobTest[DummyPipeline.type ]
+      .counters(_.size shouldBe 4)
+      .counter(
+        ScioMetrics.counter(
+          "com.spotify.elitzur.TestClasses.SeqTest",
+          "t/Seq[CountryCodeTesting]/ElitzurInvalid"
```

Comment:
IMO We shouldn't leak the underlying type, we should treat all repeated fields as the same in the reporting. Also I think this might break our current UI internally so we need to verify how it plays with that. I think in our other tools we use a placeholder for arrays and repeated fields.

Reply:
Thinking more it might be better to just report these as `CountryCodeTesting` etc and do away with the array logic. Part of my reasoning is that we are technically validating the individual pieces, not the whole Seq. As an example a validation type could validate an Int Vector holistically rather than each Int individually

Reply:
This ^ is basically what I'm trying to solve but built into elitzur. Being able to say something like `v: List[NonNegativeLong]` and just report a counter for it. Basically we just needed to report counters here https://github.com/spotify/elitzur/blob/master/elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala#L302 since the collection types are not instances of `FieldValidator[_]`. Since the collection types are not instances of field validator, I couldn't find a nice way to do the equivalent `accessor.validator.asInstanceOf[FieldValidator[_]].validationType` to get the validation type for collection type, this reason why I moved to validationType to the Validator trait. 


But given everything you said I feel like I could've just done accessor.validator.asInstanceOf[SeqLikeValidator[_]].validationType and simply record the counters like this https://github.com/spotify/elitzur/blob/master/elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala#L279-L299

Reply:
Basically was this all that was needed 
```scala
          } else {
            val v = accessor.validator.validateRecord(Unvalidated(accessor.value))
            val validationType = accessor.validator.asInstanceOf[SeqLikeValidator].validationType

            val c = config.fieldConfig(name)
            val v = accessor.validator.validateRecord(
              Unvalidated(accessor.value),
              new JStringBuilder(path.length + accessor.label.length + 1)
                .append(path).append(accessor.label).append(".").toString,
              Some(outermostClassName),
              config
            )

            if (v.isValid) {
              if (c != NoCounter) {
                reporter.reportValid(
                  outermostClassName,
                  name,
                  validationType
                )
              }
            } else if (v.isInvalid) {
              if (c == ThrowException) {
                throw new DataInvalidException(
                  s"Invalid value ${v.forceGet.toString} found for field $path${accessor.label}")
              }
              if (c != NoCounter) {
                reporter.reportInvalid(
                  outermostClassName,
                  name,
                  validationType
                )
              }
            }
            v
          }
```

ignoring the syntax errors lol

Reply:
val validationType = accessor.validator.asInstanceOf[SeqLikeValidator].validationType should return the collection item type name like you said in the other comment since it shouldnt matter that we validating T or Seq[T]